src/scales/scale.time.js

```diff
@@ -281,6 +281,29 @@ module.exports = function(Chart) {
 				max = ticks.length ? ticks[ticks.length - 1] : max;
 			}
 
+			// If the chart has bars, shift min and max based on the sample size
```

Comment:
Good point! As each tick interval can have different value within the same scale, the sample size of bars can also very. So, I removed the ruler code from controller.bar.js, and made calculateBarIndexPixels() call getBarPixelsForIndex() to get the left and right edge of the bar for a specified index. I also modified the calculation logic in scale.time.js to support `ticks.mode` and `ticks.source`.